futures-util/src/io/allow_std.rs

```diff
@@ -130,3 +128,25 @@ impl<T> AsyncSeek for AllowStdIo<T> where T: io::Seek {
         Poll::Ready(Ok(try_with_interrupt!(self.0.seek(pos))))
     }
 }
+
+impl<T> io::BufRead for AllowStdIo<T> where T: io::BufRead {
+    fn fill_buf(&mut self) -> io::Result<&[u8]> {
+        self.0.fill_buf()
+    }
+    fn consume(&mut self, amt: usize) {
+        self.0.consume(amt)
+    }
+}
+
+impl<T> AsyncBufRead for AllowStdIo<T> where T: io::BufRead {
+    fn poll_fill_buf<'a>(self: Pin<&'a mut Self>, _: &mut Context<'_>)
+        -> Poll<io::Result<&'a [u8]>>
+    {
+        // FIXME: We should use `try_with_interrupt` here.
+        Poll::Ready(io::BufRead::fill_buf(&mut self.get_mut().0))
```

Comment:
I wanted to use `try_with_interrupt!` macro here but there was a lifetime error.

```text
error[E0499]: cannot borrow `self` as mutable more than once at a time
   --> futures-util\src\io\allow_std.rs:147:44
    |
142 |     fn poll_fill_buf<'a>(mut self: Pin<&'a mut Self>, _: &mut Context<'_>)
    |                      -- lifetime `'a` defined here
...
147 |         Poll::Ready(Ok(try_with_interrupt!(self.0.fill_buf())))
    |         -----------------------------------^^^^----------------
    |         |                                  |
    |         |                                  mutable borrow starts here in previous iteration of loop
    |         returning this value requires that `self` is borrowed for `'a`
```

Reply:
oh holy cow [that's incredibly annoying](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=40a691ad15204739454d942a540bac50)

You can do it with `unsafe`: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f3c6fa0b2a118e31420de40e90f64d90

Reply:
Nice, thanks!

Reply:
@nikomatsakis says this is fixed by polonious XD

Reply:
Great, I'll look forward to that!